File: leetcode/Python/implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.st, self.t=[], []
        
    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.st.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        self.peek()
        return self.t.pop()
        
    def peek(self) -> int:
        """
        Get the front element.
        """
        if not self.t:
            while self.st:
                self.t.append(self.st.pop())
        return self.t[-1]
        
    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        return not self.st and not self.t


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: leetcode/C++/implement-queue-using-stacks.cpp
class MyQueue {
public:
    /** Initialize your data structure here. */
    stack<int> st,t;
    MyQueue() {
        
    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        st.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        int n = peek();
        t.pop();
        return n;
    }
    
    /** Get the front element. */
    int peek() {
        if(!t.empty()) return t.top();
        while(!st.empty()){
            t.push(st.top());
            st.pop();
        }
        return t.top();
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return st.empty()&&t.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */